development/tools/pkg/ipcleaner/remover.go

```diff
@@ -0,0 +1,70 @@
+package ipcleaner
+
+import (
+	"context"
+	"time"
+
+	"github.com/kyma-project/test-infra/development/tools/pkg/common"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+//go:generate mockery -name=ComputeAPI -output=automock -outpkg=automock -case=underscore
+
+// ComputeAPI abstracts access to Compute API in GCP
+type ComputeAPI interface {
+	RemoveIP(ctx context.Context, project, region, name string) error
+}
+
+// IPRemover deletes IPs provisioned by gke-long-lasting prow jobs.
+type IPRemover struct {
+	computeAPI  ComputeAPI
+	maxAttempts uint
+	backoff     uint
+	makeChanges bool
+}
+
+// New returns a new instance of IPRemover
+func New(computeAPI ComputeAPI, maxAttempts, backoff uint, makeChanges bool) *IPRemover {
+	if maxAttempts < 1 {
+		maxAttempts = 1
+	}
+	return &IPRemover{computeAPI, maxAttempts, backoff, makeChanges}
+}
+
+// Run executes ip removal process for specified IP
+func (ipr *IPRemover) Run(project, region, ipName string) error {
+	common.Shout("Trying to delete IP with name \"%s\" in project \"%s\", available in region \"%s\"", ipName, project, region)
+
+	var msgPrefix string
+	if !ipr.makeChanges {
+		msgPrefix = "[DRY RUN] "
+	}
+
+	ctx := context.Background()
+
+	var err error
+	backoff := ipr.backoff
+	for attempt := uint(0); attempt < ipr.maxAttempts; attempt = attempt + 1 {
+		var removalErr error
+		if ipr.makeChanges {
+			removalErr = ipr.computeAPI.RemoveIP(ctx, project, region, ipName)
+		}
+		if removalErr == nil {
+			log.Errorf("%sRequested deletion of IP with name \"%s\" in region \"%s\"", msgPrefix, ipName, region)
+			return nil
+		}
+		if removalErr.Error() == ipDeletionFailed {
```

Comment:
It is never a good idea to compare error messages, especially as messages are so generic like this one - `delete failed`. You should rather create your own errors like `EntryNotExist` or something like that. Simply create `errors` pkg with errors that will be used across cleaners, collectors, etc. in this project. Here is a simple example of this approach: https://github.com/asyncapi/converter-go/blob/master/pkg/error/error.go